internal/pkg/csumreader/csumreader.go

```diff
@@ -0,0 +1,91 @@
+package csumreader
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+const (
+	ChecksumLen = 8
+)
+
+var (
+	ErrChecksumMismatch = errors.New("checksum mismatch")
+)
+
+type ChecksumReader struct {
+	dataFile              io.Reader
+	checksumFile          io.Reader
```

Comment:
Actually csumreader is able to handle non-file readers.

```suggestion
	dataReader              io.Reader
	checksumReader          io.Reader
```